### PR TITLE
HHH-18719 Previous row state reuse can provide detached entities to the consumer

### PR DESCRIPTION
**Pull request by reda-alaoui**
Fix for https://hibernate.atlassian.net/browse/HHH-18719

----------------------
By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt)
and can be relicensed under the terms of the [LGPL v2.1 license](https://www.gnu.org/licenses/old-licenses/lgpl-2.1.txt) in the future at the maintainers' discretion.
For more information on licensing, please check [here](https://github.com/hibernate/hibernate-orm/blob/main/CONTRIBUTING.md#legal).

----------------------

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/sql/results/graph/entity/internal/EntityInitializerImpl.java [hunk 1]
```diff
@@ -588,7 +590,6 @@ && areKeysEqual( oldEntityKey.getIdentifier(), id ) ) {
 			if ( !entityKeyOnly ) {
 				notifySubInitializersToReusePreviousRowInstance( data );
 			}
-			return;
```
Comment (dreab8) [inline hibernate-core/src/main/java/org/hibernate/sql/results/graph/entity/internal/EntityInitializerImpl.java:591]: why removing `return`?

Comment (reda-alaoui) [reply hibernate-core/src/main/java/org/hibernate/sql/results/graph/entity/internal/EntityInitializerImpl.java]: It is a mistake. Good catch.

**Comment by dreab8** [comment]
Hi @reda-alaoui , 

Thanks a lot for your work, 
with @beikov we are trying to see if we can avoid the map lookup overhead.

**Comment by beikov** [comment]
Thanks. Superseded by https://github.com/hibernate/hibernate-orm/pull/9102

**Pull request closed without merging**